File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/school/School.kt
package com.nevexis.backend.schoolManagement.school

import kotlinx.serialization.Serializable


@Serializable
data class School(
    val id: Int,
    val schoolName: String,
    val city: String,
    val address: String,
    val rooms: List<RoomToSubjects>? = null
)

@Serializable
data class RoomToSubjects(
    val room: String,
    val subjects: List<String>
)

File: frontend/src/model/School.ts
export interface School {
    id: number,
    schoolName: string,
    city: string,
    address: string
    rooms: RoomToSubjects[] | null
}

export interface RoomToSubjects {
    room: string
    subjects: string[]
}

export const roomToSubjectsText = (roomToSubjects: RoomToSubjects) => {
    return `${roomToSubjects.room} ${roomToSubjects.subjects.length > 0 ? ` - Стая за часове по: ${roomToSubjects.subjects.join(',')}` : ''}`
}